query-engine/driver-adapters/js/connector-test-kit-executor/src/index.ts

```diff
@@ -170,10 +181,39 @@ function respondOk(requestId: number, payload: unknown) {
 }
 
 async function initQe(url: string, prismaSchema: string, logCallback: qe.QueryLogCallback): Promise<[engines.QueryEngineInstance, ErrorCapturingDriverAdapter]> {
-    const pool = new pgDriver.Pool({ connectionString: url })
-    const adapter = bindAdapter(new pg.PrismaPg(pool))
-    const engineInstance = qe.initQueryEngine(adapter, prismaSchema, logCallback)
-    return [engineInstance, adapter];
+    const adapter = await adapterFromEnv(url) as DriverAdapter
+    const errorCapturingAdapter = bindAdapter(adapter)
+    const engineInstance = qe.initQueryEngine(errorCapturingAdapter, prismaSchema, logCallback)
+    return [engineInstance, errorCapturingAdapter];
+}
+
+async function adapterFromEnv(url: string): Promise<DriverAdapter> {
+    const adapter = `${process.env.DRIVER_ADAPTER as string}`
+
+    if (adapter == '') {
+        throw new Error("DRIVER_ADAPTER is not defined or empty.");
```

Comment:
`adapter` will be a literal `"undefined"` string if `DRIVER_ADAPTER` is not defined, you need to change line 191 to something like

```ts
const adapter = process.env.DRIVER_ADAPTER ?? ''
```

for this check to work this way.